packages/firebase_remote_config/test/firebase_remote_config_test.dart

```diff
@@ -0,0 +1,163 @@
+    });
+  });
+
+  group('$RemoteConfig', () {
+    const MethodChannel channel = const MethodChannel(
+      'plugins.flutter.io/firebase_remote_config',
+    );
+
+    final List<MethodCall> log = <MethodCall>[];
+
+    final int lastFetchTime = 1520618753782;
+    RemoteConfig remoteConfig;
+
+    setUp(() async {
+      channel.setMockMethodCallHandler((MethodCall methodCall) async {
+        log.add(methodCall);
+        switch (methodCall.method) {
+          case 'RemoteConfig#fetch':
+            return <String, dynamic>{
+              'LAST_FETCH_TIME': lastFetchTime,
+              'LAST_FETCH_STATUS': 'success',
+            };
+          case 'RemoteConfig#instance':
+            return getDefaultInstance();
+          case 'RemoteConfig#activate':
+            return <String, dynamic>{
+              'parameters': <String, dynamic>{
+                'param1': <String, dynamic>{
+                  'source': 'static',
+                  'value': <int>[118, 97, 108, 49], // UTF-8 encoded 'val1'
+                },
+                'param2': <String, dynamic>{
+                  'source': 'static',
+                  'value': <int>[49, 50, 51, 52, 53], // UTF-8 encoded '12345'
+                },
+                'param3': <String, dynamic>{
+                  'source': 'default',
+                  'value': <int>[51, 46, 49, 52], // UTF-8 encoded '3.14'
+                },
+                'param4': <String, dynamic>{
+                  'source': 'static',
+                  'value': <int>[116, 114, 117, 101] // UTF-8 encoded 'true'
+                }
+              },
+              'newConfig': true,
+            };
+          case 'RemoteConfig#setConfigSettings':
+            return null;
+          default:
+            return true;
+        }
+      });
+      remoteConfig = await RemoteConfig.instance;
+      log.clear();
+    });
+
+    test('fetch', () async {
+      await remoteConfig.fetch(expiration: const Duration(hours: 1));
+      expect(
+        log,
+        <Matcher>[
+          isMethodCall(
+            'RemoteConfig#fetch',
+            arguments: <String, dynamic>{
+              'expiration': 3600,
+            },
+          ),
+        ],
+      );
+    });
+
+    test('activate', () async {
+      final bool newConfig = await remoteConfig.activateFetched();
+      expect(
+        log,
+        <Matcher>[
+          isMethodCall(
+            'RemoteConfig#activate',
+            arguments: null,
+          ),
+        ],
+      );
+      expect(newConfig, true);
+      expect(remoteConfig.getString('param1'), 'val1');
+      expect(remoteConfig.getInt('param2'), 12345);
+      expect(remoteConfig.getDouble('param3'), 3.14);
+      expect(remoteConfig.getBool('param4'), true);
+    });
+
+    test('setConfigSettings', () async {
+      expect(remoteConfig.remoteConfigSettings.debugMode, true);
+      final RemoteConfigSettings remoteConfigSettings =
+          new RemoteConfigSettings();
```

Comment:
Use the constructor to initialize `debugMode`

Reply:
Done